src/overlays/actors/ovl_Bg_Dkjail_Ivy/z_bg_dkjail_ivy.c

```diff
@@ -28,47 +36,175 @@ const ActorInit Bg_Dkjail_Ivy_InitVars = {
+
+            func_800B12F0(play, &pos, &gZeroVec3f, &accel, ((Rand_Next() >> 26) + 10), (Rand_Next() >> 28), 50);
+        }
+    }
+}
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80ADE950;
-extern InitChainEntry D_80ADE98C[];
-
-extern UNK_TYPE D_06000080;
-extern UNK_TYPE D_060011A8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE230.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE6AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE6C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE734.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE748.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE7E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/func_80ADE7F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dkjail_Ivy/BgDkjailIvy_Draw.s")
+void BgDkjailIvy_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    BgDkjailIvy* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    Collider_InitCylinder(play, &this->collider);
+
+    if (Flags_GetSwitch(play, BG_DKJAIL_GET_SWITCH(thisx))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        DynaPolyActor_LoadMesh(play, &this->dyna, &object_dkjail_obj_Colheader_0011A8);
+        Collider_SetCylinder(play, &this->collider, &this->dyna.actor, &sCylinderInit);
+        Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+        this->alpha = 255;
+        BgDkjailIvy_SetupSetCollision(this);
+    }
+}
+
+void BgDkjailIvy_Destroy(Actor* thisx, PlayState* play) {
+    BgDkjailIvy* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void BgDkjailIvy_SetupSetCollision(BgDkjailIvy* this) {
+    this->actionFunc = BgDkjailIvy_SetCollision;
+}
+
+void BgDkjailIvy_SetCollision(BgDkjailIvy* this, PlayState* play) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        this->dyna.actor.flags |= ACTOR_FLAG_10;
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+        BgDkjailIvy_SetupCutscene(this);
+    } else {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
```

Comment:
Really this state is `Idle` or `Wait`: it's what it'll do before it's interacted with.